packages/model/addon/-private/record-state.ts

```diff
@@ -0,0 +1,391 @@
+import { assert } from '@ember/debug';
+import { dependentKeyCompat } from '@ember/object/compat';
+import { cached, tracked } from '@glimmer/tracking';
+
+import notifyChanges from './notify-changes';
+
+type RecordData = import('@ember-data/record-data/-private').RecordData;
+type RequestCache = import('@ember-data/store/-private/system/request-cache').default;
+type Model = InstanceType<typeof import('./model')>;
+type StableRecordIdentifier = import('@ember-data/store/-private/ts-interfaces/identifier').StableRecordIdentifier;
+type CoreStore = import('@ember-data/store/-private/system/core-store').default;
+
+function isInvalidError(error) {
+  return error && error.isAdapterError === true && error.code === 'InvalidError';
+}
+
+class Tag {
+  declare rev: number;
+  declare isDirty: boolean;
+  declare value: any;
+
+  constructor() {
+    this.rev = 1;
+    this.isDirty = true;
+    this.value = undefined;
+  }
+  @tracked _tracking = 0;
```

Comment:
Do we need a tracked property for this data structure?  I assumed I would find a getter somewhere that is consuming it.  It seems most of the semantics here are preventing running the tagged property from running/not running if `isDirty`, which is accomplished through an explicit `notify`.  If I am missing something, what is `_tracking` forcing to recompute?

Reply:
It's consumed by subscribe which we call on all accesses to a property decorated with @tagged. This consumption is what tells glimmer to recompute once we've decided to notify.

Reply:
Oh right.  This is the first time I have seen a function call to setup tracking.  The tracking system must be able to "see through" that function call and recompute the original accessor when `_tracking` is updated